cmd/rotate_certs.go

```diff
@@ -149,7 +157,13 @@ func (rcc *rotateCertsCmd) run(cmd *cobra.Command, args []string) error {
 	log.Infoln("Generating new certificates")
 
 	// reset the certificateProfile and use the exisiting certificate generation code to generate new certificates.
-	rcc.containerService.Properties.CertificateProfile = &api.CertificateProfile{}
+	ca := rcc.containerService.Properties.CertificateProfile.CaCertificate
```

Comment:
Is there a chance CaCertificate or CaPrivateKey will be empty?

Reply:
The first step initialy was to create a new empty certificate profile. This meant that the previous CA certs will not be present and therefore regenerated. This is why i temporarily store the CA details before running the certificate generation